connector/connect/common/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -379,6 +380,15 @@ message ScalarScalaUDF {
   bool nullable = 4;
 }
 
+message ScalaUDAF {
+  // (Required) Serialized JVM object of the Aggregator instance, including buffer and output encoders
+  bytes payload = 1;
+  // (Required) Input type of the UDAF
+  DataType inputType = 2;
+  // (Required) True if the UDAF can return null value
+  bool nullable = 3;
```

Comment:
Why is the return type not needed?

Reply:
For the record we should not rely on the payload for this.

Reply:
Added back!